src/executor/vm/tests.rs

```diff
@@ -25,42 +25,47 @@ fn is_send() {
     test::<super::VirtualMachinePrototype>();
 }
 
-// TODO: test below should run for both wasmi and wasmtime
-
 #[test]
 fn basic_seems_to_work() {
-    let module = super::Module::new(
-        &include_bytes!("./test-polkadot-runtime-v9160.wasm")[..],
-        super::ExecHint::CompileAheadOfTime,
-    )
-    .unwrap();
-
-    let prototype = super::VirtualMachinePrototype::new(&module, |_, _, _| Ok(0)).unwrap();
-
-    // Note that this test doesn't test much, as anything elaborate would require implementing
-    // the Substrate/Polkadot allocator.
-
-    let mut vm = prototype
-        .start(
-            super::HeapPages::new(1024),
-            "Core_version",
-            &[super::WasmValue::I32(0), super::WasmValue::I32(0)],
+    fn test(exec_hint: super::ExecHint) {
+        let module = super::Module::new(
+            &include_bytes!("./test-polkadot-runtime-v9160.wasm")[..],
+            exec_hint,
         )
         .unwrap();
 
-    loop {
-        match vm.run(None) {
-            Ok(super::ExecOutcome::Finished {
-                return_value: Ok(_),
-            }) => break,
-            Ok(super::ExecOutcome::Finished {
-                return_value: Err(_),
-            }) => panic!(),
-            Ok(super::ExecOutcome::Interrupted { id: 0, .. }) => break,
-            Ok(super::ExecOutcome::Interrupted { .. }) => panic!(),
-            Err(_) => panic!(),
+        let prototype = super::VirtualMachinePrototype::new(&module, |_, _, _| Ok(0)).unwrap();
+
+        // Note that this test doesn't test much, as anything elaborate would require implementing
+        // the Substrate/Polkadot allocator.
+
+        let mut vm = prototype
+            .start(
+                super::HeapPages::new(1024),
+                "Core_version",
+                &[super::WasmValue::I32(0), super::WasmValue::I32(0)],
+            )
+            .unwrap();
+
+        loop {
+            match vm.run(None) {
+                Ok(super::ExecOutcome::Finished {
+                    return_value: Ok(_),
+                }) => break,
+                Ok(super::ExecOutcome::Finished {
+                    return_value: Err(_),
+                }) => panic!(),
```

Comment:
do you want to panic with error?

```suggestion
                }) => panic!("{:?}", e),
```

Reply:
Well, it's just a test :shrug:

Reply:
Okay